grails-app/assets/javascripts/directives/streama-video-player-directive.js

```diff
@@ -194,9 +196,22 @@ streamaApp.directive('streamaVideoPlayer', [
           }
         };
 
-        video.ontimeupdate = function(){
+        /** Get CurrentEpisode/Skip on IntroStart to IntroEnd if skipIntro == true*/
+        var currEpisode = null;
+        video.ontimeupdate = function(event){
           $scope.currentTime = video.currentTime;
           $scope.$apply();
+       if(skipIntro)
+          {
+            if(currEpisode == null)
+            {
+              currEpisode = playerService.getVideoOptions().currentEpisode;
```

Comment:
instead of checking and setting the currEpisode here, you should check it in the onCanPlay method as everything gets initialized there.